### ISSUE TITLE
Unable to import some notes even after selecting "Merge notetypes" in the Import File window

### ISSUE THREAD

**Issue by nickel8448** [ankitects/anki#2693]
(Filing this request as per dae's comment on the forum - https://forums.ankiweb.net/t/anki-23-10-beta/34912/85)

Notes are not being merged even when the “Merge notetypes” option was enable. I hope I am using the feature right, please let me know if this is not the intended way to use it.

[apkg to import zip file](https://github.com/ankitects/anki/files/12762834/edited.27.09.2023.apkg.zip)
[deck where it needs to be imported](https://github.com/ankitects/anki/files/12762879/Filtered.Deck.10_06.apkg.zip)

**Comment by RumovZ** [comment]
Thanks, that helped a lot.

@dae, I'm afraid there are at least three cases, where a note will currently continue to conflict:
1. The exporter changed the notetype of a note.
2. The importer changed the notetype of a note.
3. The notetype was remapped during an import with a previous release which didn't store the original notetype id.

Obviously, the big one is 3, which we should be supporting. We could, if we encounter a note with a conflicting notetype and `merge notetypes` is enabled, merge the note's current notetype into the incoming notetype as well. But unfortunately, we can't know if we're dealing with case 1, 2, or 3.
What are your thoughts? Should we merge these notetypes? Or do you see a better way?

Also, I've found another bug while investigating this and will push a fix.

**Comment by dae** [comment]
I'm not very up-to-speed with this code at the moment, so I don't have a good feel for the potential implications here. Since we can't differentiate between 1-3, what downsides can you foresee from just merging the notetypes in such cases? I suspect the typical user just wants the material imported, and is not fussed about what happens to their local notetypes, provided it doesn't break future imports or cause them to lose data.

**Comment by RumovZ** [comment]
Let's assume a user deliberately changed the notetype of an imported note X from A to B. Then upon importing an update (merging enabled), B would be merged into A. X would be changed back to A, but also any notes the user had originally created with B.
I guess while this may not be the best action to take here, it also shouldn't be too surprising to the user. As for dataloss, templates of B would be overwritten with A's *if* they have the same name and A was updated more recently.

The reverse case, where the exporter changed the notetype, is probably even less critical, as to the importing user that's not different to when the exporter drastically modifies the notetype directly. (Which they shouldn't do anyway.)

So the downsides seem limited, and changing notetypes of shared notes is hopefully rare anyway. The implementation might be a bit messy, though, as now we have to check the notetypes of incoming notes before we can start importing notetypes.

**Comment by dae** [comment]
That sounds reasonable, as the user has requested merging after all. If it's practical to implement, let's go ahead with that.
